lib/Builder/Copier.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\RST\Builder;
+
+use Symfony\Component\Filesystem\Filesystem;
+use function basename;
+use function dirname;
+use function is_dir;
+
+class Copier
+{
+    /** @var Filesystem */
+    private $filesystem;
+
+    /** @var string[][] */
+    private $toCopy = [];
+
+    /** @var string[] */
+    private $toMkdir = [];
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    public function doCopy(string $sourceDirectory, string $targetDirectory) : void
+    {
+        foreach ($this->toCopy as $copy) {
+            [$source, $destination] = $copy;
+
+            if ($source[0] !== '/') {
+                $source = $sourceDirectory . '/' . $source;
+            }
+
+            $destination = $targetDirectory . '/' . $destination;
+
+            if (is_dir($source) && is_dir($destination)) {
+                $destination = dirname($destination);
+            }
+
+            if (is_dir($source)) {
+                $this->filesystem->mirror($source, $destination);
+            } else {
+                $this->filesystem->copy($source, $destination);
+            }
+        }
```

Comment:
Should we clear the `$this->toCopy` when the method is finished? (same below for toMkdir)?